pkg/signup/service/signup_service.go

```diff
@@ -223,15 +234,28 @@ func (s *ServiceImpl) reactivateUserSignup(ctx *gin.Context, existing *toolchain
 // GetSignup returns Signup resource which represents the corresponding K8s UserSignup
 // and MasterUserRecord resources in the host cluster.
 // Returns nil, nil if the UserSignup resource is not found or if it's deactivated.
-func (s *ServiceImpl) GetSignup(userID string) (*signup.Signup, error) {
+func (s *ServiceImpl) GetSignup(userID, username string) (*signup.Signup, error) {
+	// Declare variables up front
+	var userSignup *toolchainv1alpha1.UserSignup
+	var err error
 
-	// Retrieve UserSignup resource from the host cluster
-	userSignup, err := s.CRTClient().V1Alpha1().UserSignups().Get(EncodeUserID(userID))
+	// Retrieve UserSignup resource from the host cluster, using the specified UserID
+	userSignup, err = s.CRTClient().V1Alpha1().UserSignups().Get(EncodeUserIdentifier(userID))
 	if err != nil {
+		// If the UserSignup was not found, attempt to find it using the specified username instead
 		if apierrors.IsNotFound(err) {
-			return nil, nil
+			userSignup, err = s.CRTClient().V1Alpha1().UserSignups().Get(EncodeUserIdentifier(username))
+
+			// Otherwise if there was any other error, return it instead
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					return nil, nil
+				}
+				return nil, err
```

Comment:
Same here. Let's add a test for this error.

Reply:
Done